### PR TITLE
feat: PWA support

### LINKED ISSUE #257: [feature]: Add PWA function
### 1~3 main use cases of the proposed feature

Make it more like a desktop application.
Send shortcut to desktop, and open it in a browser container.

### Ideas for solution

PWA(Progressive Web App)

### what types of users can benefit from using your proposed feature

more like open the shotcut than type a url to open a product users

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pingren**
Will close #257

https://user-images.githubusercontent.com/5123601/197326132-e4ce040f-0a1c-471d-aabf-3ddf314e20f4.mp4

**NOTES**
If you like to test it with `pnpm start` locally, please git checkout to 1cabbf9cc6a895f3319e5b8533ac77e7df9c88c8.

### CODE REVIEW

File: apps/ligo-virgo/src/assets/manifest.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "display": "standalone",
+    "icons": [
+        {
+            "sizes": "192x192",
```
Comment (linonetwo) [inline apps/ligo-virgo/src/assets/manifest.json:5]: I think it is better to have 256x256 and 144x144, as recommended in the pwa documentation before. Don't know if this changed these years.

File: apps/ligo-virgo/webpack.config.js [hunk 2]
```diff
@@ -181,6 +183,38 @@ module.exports = function (webpackConfig) {
...
+        isProd &&
+            !isE2E &&
+            new WorkboxPlugin.GenerateSW({
+                // these options encourage the ServiceWorkers to get in there fast
+                // and not allow any straggling "old" SWs to hang around
+                clientsClaim: true,
+                skipWaiting: true,
+                runtimeCaching: [
+                    {
+                        handler: 'NetworkFirst',
```
Comment (linonetwo) [inline apps/ligo-virgo/webpack.config.js:213]: I think it is better to use StaleWhileRevalidate for HTML and JS, and CacheFirst for image

Here is how I did it in my blog, just my opinion FYI https://github.com/linonetwo/wiki/blob/27608d25ac955dfb4670f63cdbf0da5ba837e6ec/public/service-worker.js

File: apps/ligo-virgo/src/index.html [hunk 3]
```diff
@@ -6,9 +6,6 @@
         <meta name="viewport" content="width=device-width, initial-scale=1" />
         <link rel="icon" href="/favicon.ico" />
         <title>AFFiNE - All In One Workos</title>
-        <script>
-            window.global = window;
```
Comment (linonetwo) [inline apps/ligo-virgo/src/index.html:10]: What did this do?

**Comment by linonetwo** [comment]
Thank you for this PR, welcome!

**Comment by linonetwo** [comment]
This branch is for pre-alpha, actually, we are working on a Alpha version now, in the anmaster branch.

File: apps/ligo-virgo/webpack.config.js (hunk 2 above)
Comment (pingren) [reply apps/ligo-virgo/webpack.config.js:213]: Great suggestions. I will update.

**Review by pingren — COMMENTED** [review]
Thank you for review. I will update the PR when I have time.

File: apps/ligo-virgo/src/assets/manifest.json (hunk 1 above)
Comment (linonetwo) [reply apps/ligo-virgo/src/assets/manifest.json:5]: No, just use larger one of this icon is ok

**Review by Toast063 — COMMENTED** [review]
Approve

**Comment by pingren** [comment]
@linonetwo PR is updated based on your comments. Also, I noticed #655 is merged using `next-pwa` so I am not sure if this PR is still worth to merge. You can make the decision anyway.

**Comment by linonetwo** [comment]
The pre-alpha still matters for our community, so this PR is welcome, let me merge it, thank you.

**Review by linonetwo — APPROVED** [review]

**Comment by linonetwo** [comment]
Welcome to add more things that you are interested in.

**Pull request merged**